core/src/main/java/me/lokka30/treasury/plugin/core/utils/UpdateChecker.java

```diff
@@ -38,18 +41,102 @@ public static void checkForUpdates() {
                 try (Reader in = new InputStreamReader(url.openStream())) {
                     JsonObject object = Utils.GSON.fromJson(in, JsonObject.class);
                     String latestVersion = object.get("current_version").getAsString();
-                    if (!plugin.getVersion().equalsIgnoreCase(latestVersion)) {
-                        plugin.logger().warn(
-                                "A new Treasury update is available - '&bv"
-                                        + latestVersion
-                                        + "&7' - please update as soon as possible."
-                                        + " &8(&7You're running '&bv" + plugin.getVersion() + "&7'&8)"
-                        );
-                    }
+                    handleVersionCheck(latestVersion);
                 }
```

Comment:
We should also catch `NetException` (if I remember  correctly), which fires if the user's server couldn't connect

Reply:
nowhere in java's documentation specifies that anything like that gets thrown. fairly sure catching only the IOException is safe.

Reply:
I apologize, I was thinking of the update checker from the Spigot Wiki. All good.